src/main/java/controllers/UserProfileController.java

```diff
@@ -87,128 +88,164 @@ else if (choice == 1) {
+                this.inOut.setOutput("Successfully unfollowed the target user!");
+            } else {
+                this.inOut.setOutput("User is not followed to begin with!");
+            }
+        } catch (IOException e) {
+            inOut.setOutput("There was an error: " + e);
+        }
+    }
+
+    /**
+     * Display all currently registered usernames
      */
-    private void runUnfollowUser(User user, User targetUser) {
-        if (this.userManager.unfollowUser(user, targetUser)) {
-            this.inOut.setOutput("Successfully unfollowed the target user!");
+    private void runDisplayAllUsers() {
+        User[] users = this.databaseManager.getAllUsers();
+        StringBuilder allUsernames = new StringBuilder();
+        for (User user : users) {
+            allUsernames.append(" ").append(this.userManager.getUsername(user)).append(", ");
         }
-        else {
-            this.inOut.setOutput("User is not followed to begin with!");
+        this.inOut.setOutput("Currently registered users:" + allUsernames.substring(0, allUsernames.length() - 2));
+    }
+
+    /**
+     * Display user's posts
+     */
+    private void runDisplayUserPosts(User user) {
+        PostController postController = new PostController(this.inOut, this.databaseManager, this.loginManager);
+        PostManager postManager = new PostManager(this.userManager.getUserPosts(user));
+        for(Post post: this.userManager.getUserPosts(user)) {
+            postController.displayPost(postManager.getPostId(post));
+        }
+    }
+
+    /**
+     * Run the customize username logic
+     */
+    private void runCustomizeUsername() {
+        try {
+            String newUsername = this.inOut.getInput("Enter what you'd like your new username to be: ");
+            runChangeUsernameDisplay(this.loginManager.getCurrUser(), newUsername);
+        } catch (IOException e) {
+            inOut.setOutput("There was an error: " + e);
         }
     }
 
     /**
-     * Display user's following list
+     * Run the customize password logic
      */
-    private String displayFollowingList(User user) {
-        ArrayList<String> followingListUsernames = this.userManager.getFollowingListUsernames(user);
-        String followingList = "";
-        for (String username: followingListUsernames) {
-            followingList = followingList.concat(username + ", ");
+    private void runCustomizePassword() {
+        try {
+            String newPassword = this.inOut.getInput("Enter what you'd like your new password to be: ");
+            runChangePasswordDisplay(this.loginManager.getCurrUser(), newPassword);
+        } catch (IOException e) {
+            inOut.setOutput("There was an error: " + e);
         }
-        return followingList;
     }
 
+    /**
+     * Run the customize bio logic
+     */
+    private void runCustomizeBio() {
+        try {
+            String newBio = this.inOut.getInput("Enter what you'd like your new bio to be: ");
```

Comment:
I think you should add the option to return to main menu in most if not all these "run" methods

Reply:
yup sounds good!